packages/components/fields/select-field/src/select-field.tsx

```diff
@@ -0,0 +1,370 @@
+import {
+  Component,
+  isValidElement,
+  type ReactElement,
+  type ReactNode,
+  type FocusEventHandler,
+} from 'react';
+import {
+  createSequentialId,
+  filterDataAttributes,
+  getFieldId,
+  warning,
+} from '@commercetools-uikit/utils';
+import Constraints from '@commercetools-uikit/constraints';
+import Spacings from '@commercetools-uikit/spacings';
+import FieldLabel from '@commercetools-uikit/field-label';
+import SelectInput from '@commercetools-uikit/select-input';
+import FieldErrors from '@commercetools-uikit/field-errors';
+import type { Props as ReactSelectProps } from 'react-select';
+
+type TErrorRenderer = (key: string, error?: boolean) => ReactNode;
+type TOption = {
+  value: string;
+};
+declare type TOptionObject = {
```

Comment:
Nit: I'm wondering if we need the `declare` syntax here. Same applies to line 28